### PR TITLE
Remove 'coreconcepts-' prefix from styles, admonition blocks, etc

### PR DESCRIPTION
**Pull request by pdaoust**
There are some template elements called `coreconcepts-intro`, `coreconcepts-orientation`, etc. The naming is a legacy from when they were only used on the core concepts pages, but now they're getting used in other places too.

There are a couple places where they show up:

* The Markdown parser has been extended to recognise them as special blocks that deserve their own styling
* There are matching CSS styles

~~This is just a simple search-and-replace.~~ _Update: see below_. I'm pretty sure the shortened names don't clash with anything.

## Update

I found out the styles weren't hooked up to these block containers anyway, so I've done some work to make them work. You can see all three at work in e.g., https://deploy-preview-438--developer-portal-production.netlify.app/concepts/7_validation/

**One big change** is that the big `::: storysequence` block has been changed to more verbose, but much simpler and less magical, `::: storystep` blocks around each step in a story sequence. I don't think we'd ever reimplemented the original site's behavour, in which each step was a list item.

### CODE REVIEW

**Comment by pdaoust** [comment]
okay, this is weird... I just discovered that these special Markdown blocks get custom markup and styles, but never the twain do meet -- the styles live in a file that never gets loaded by any page!

File: src/scss/_blocks.scss [hunk 1]
```diff
@@ -0,0 +1,28 @@
+.intro {
+  font-size: $fs-even-bigger;
```
Comment (mattyg) [inline src/scss/_blocks.scss:2]: font feels a bit too large to me

image

File: src/scss/_blocks.scss [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  border-left: 4px solid $cl-gray;
+  padding-left: 1em;
+  font-weight: 500;
+  line-height: 1.55;
+}
+
+.orientation, .storystep {
```
Comment (mattyg) [inline src/scss/_blocks.scss:9]: I like the added background color to distinguish the content. I think the text could be distinguished a bit more clearly too? Something like this:

image

**Review by mattyg — APPROVED** [review]
Noted a few potential styling tweaks (I don't have a good design sense so feel free to ignore).

Also I see that the Get Started page and Resources page have "intro" text that isn't tagged for the intro styling.

File: src/scss/_blocks.scss (hunk 2 above)
Comment (pdaoust) [reply src/scss/_blocks.scss:9]: I might eventually apply this to all images, actually -- we're working with a graphic designer who's added a gray background to all her example illustrations and it looks spiffy. I feel like the box around the text may add too much differentiation... maybe just a little horizontal line. Tough to do this in CSS, but I might be able to.

File: src/scss/_blocks.scss (hunk 1 above)
Comment (pdaoust) [reply src/scss/_blocks.scss:2]: fair feedback... guess the large font won't look good for every intro block. I'm personally okay with this look, but if it looks jarring to you I can play with sizes.

File: src/scss/_blocks.scss (hunk 2 above)
Comment (mattyg) [reply src/scss/_blocks.scss]: sure -- you can do "border-top" on the text element for a horizontal line

File: src/scss/_blocks.scss (hunk 1 above)
Comment (mattyg) [reply src/scss/_blocks.scss]: eh nah lets leave it for a real designer down the road.

Comment (pdaoust) [reply src/scss/_blocks.scss]: I used to be one :)

File: src/scss/_blocks.scss (hunk 2 above)
Comment (pdaoust) [reply src/scss/_blocks.scss]: check out core concept seven -- I played around with `<hr/>` elements, so the author can opt in if they like the look and opt out if they don't. Could lead to inconsistency, but I'd rather inconsistency that fit the need than consistency that looked awkward in places.

https://deploy-preview-438--developer-portal-production.netlify.app/concepts/7_validation/

File: src/scss/_blocks.scss (hunk 1 above)
Comment (mattyg) [reply src/scss/_blocks.scss]: ha! well than I defer to you

File: src/scss/_blocks.scss (hunk 2 above)
Comment (pdaoust) [reply src/scss/_blocks.scss]: updated: I've applied the treatment to all story steps. (yay regexes)

Comment (mattyg) [reply src/scss/_blocks.scss]: nice - I prefer it with the original margin, i.e.
image

but also I'm not a designer and defer to you

**Pull request merged**